src/blocks/blocks/form/edit.js

```diff
@@ -300,71 +384,350 @@ const Edit = ({
+
+		/**
+		 * TODO: Refactor this based on #748
+		 */
+
+		if ( blockRef.current ) {
+			blockRef.current?.style?.setProperty( '--padding', padding( attributes.inputPadding ) );
+			blockRef.current?.style?.setProperty( '--borderRadius', px( attributes.inputBorderRadius ) );
+			blockRef.current?.style?.setProperty( '--borderWidth', px( attributes.inputBorderWidth ) );
+			blockRef.current?.style?.setProperty( '--borderColor', attributes.inputBorderColor || null );
+			blockRef.current?.style?.setProperty( '--labelColor', attributes.labelColor || null );
+			blockRef.current?.style?.setProperty( '--inputWidth', per( attributes.inputWidth ) );
+			blockRef.current?.style?.setProperty( '--submitColor', attributes.submitColor || null );
+			blockRef.current?.style?.setProperty( '--inputGap', px( attributes.inputGap ) );
+			blockRef.current?.style?.setProperty( '--inputsGap', px( attributes.inputsGap ) );
+			blockRef.current?.style?.setProperty( '--labelFontSize', px( attributes.labelFontSize ) );
+			blockRef.current?.style?.setProperty( '--submitFontSize', px( attributes.submitFontSize ) );
+		}
+	}, [ blockRef.current, attributes ]);
+
 	return (
 		<Fragment>
-			<Inspector
-				attributes={ attributes }
-				setAttributes={ setAttributes }
-			/>
+			<FormContext.Provider
+				value={{
+					savedEmail,
+					apiKey,
+					setApiKey,
+					isEmailLoaded,
+					listIDOptions,
+					setListIDOptions,
+					fetchListIdStatus,
+					setFetchListIdStatus,
+					saveFormOptions,
+					email,
+					setEmail,
+					saveIntegrationApiKey,
+					fetchApiKeyStatus,
+					savedIntegration,
+					saveIntegration,
+					savedData
+				}}
+			>
+				<Inspector
+					attributes={ attributes }
+					setAttributes={ setAttributes }
+				/>
+
+				<div { ...blockProps }>
+					{
+						( hasInnerBlocks ) ? (
+							<form
+								ref={blockRef}
+								className="otter-form__container"
+								onSubmit={ () => false }
+							>
+								<InnerBlocks
+								/>
+
+								{
+									attributes.hasCaptcha && ( ! isAPILoaded || ! isAPISaved ) && (
+										<Placeholder
+											className="otter-form-captcha"
+											isAPILoaded={ isAPILoaded }
+											isAPISaved={ isAPISaved }
+											saveAPIKey={ saveCaptchaAPIKey }
+											siteKey={ googleCaptchaAPISiteKey }
+											secretKey={ googleCaptchaAPISecretKey }
+											setSiteKey={ setGoogleCaptchaAPISiteKey }
+											setSecretKey={ setGoogleCaptchaAPISecretKey }
+										/>
+									)
+								}
 
-			<div { ...blockProps }>
-				{
-					( hasInnerBlocks ) ? (
-						<div className="otter-form__container">
-							<InnerBlocks
+								{ 'submit-subscribe' === attributes.action && (
+									<div className="otter-form-consent">
+										<input id="o-consent" name="o-consent" type="checkbox" />
+										<label htmlFor="o-consent">
+											{ __( 'I consent that my name and email to be collected.', 'otter-blocks' ) }
```

Comment:
@HardeepAsrani, we should find another sentence for this. We need something that sounds more general, like: You agree to share the data with partnered providers.

Reply:
@mghenciu any idea?

Reply:
What is this for? Need more details.

Reply:
![image](https://user-images.githubusercontent.com/17597852/163953287-c2d02051-9e93-4eb8-8116-add5deae2002.png)

This appears when you set the option in integration to subscribe to the provider mailing list (Sendinblue) and send the email. You need to give consent to be added to a third-party provider.